packages/amplify-util-uibuilder/src/commands/utils/codegenResources.ts

```diff
@@ -83,15 +88,46 @@ export const createUiBuilderTheme = (context: $TSContext, schema: StudioTheme):
   }
 };
 
+/**
+ * Writes form file to the work space
+ */
+export const createUiBuilderForm = (context: $TSContext, schema: StudioForm, dataSchema?: GenericDataSchema): StudioForm => {
+  const uiBuilderComponentsPath = getUiBuilderComponentsPath(context);
+  const rendererFactory = new StudioTemplateRendererFactory(
+    (form: StudioForm) => (new AmplifyFormRenderer(form, dataSchema, config) as unknown) as StudioTemplateRenderer<
+        unknown,
+        StudioForm,
+        FrameworkOutputManager<unknown>,
+        RenderTextComponentResponse
+      >,
+  );
+
+  const outputPathDir = uiBuilderComponentsPath;
+
+  const rendererManager = new StudioTemplateRendererManager(rendererFactory, {
+    outputPathDir,
+  });
+
+  try {
+    rendererManager.renderSchemaToTemplate(schema);
+    return schema;
+  } catch (e) {
+    printer.debug(e);
+    printer.debug('Skipping invalid form with schema');
+    printer.debug(JSON.stringify(schema, null, 2));
```

Comment:
Is there any way `JSON.stringify` can throw here?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#exceptions

Reply:
The shape should be correct since it's coming from the sdk. An error can occur on codegen side of the shape being passed has invalid references (ex. a form references a model that no longer exists) which doesn't mean it's an invalid json shape.